tests/simpleKVBC/TesterReplica/internalCommandsHandler.cpp

```diff
@@ -131,12 +129,45 @@ void InternalCommandsHandler::execute(InternalCommandsHandler::ExecutionRequests
     }
   }
 
-  if (!pre_execute && (merkleUpdates.size() > 0 || verUpdates.size() > 0)) {
+  if (merkleUpdates.size() > 0 || verUpdates.size() > 0) {
     // Write Block accumulated requests
     writeAccumulatedBlock(requests, verUpdates, merkleUpdates);
   }
 }
 
+void InternalCommandsHandler::preExecute(IRequestsHandler::ExecutionRequest &req,
+                                         std::optional<bftEngine::Timestamp> timestamp,
+                                         const std::string &batchCid,
+                                         concordUtils::SpanWrapper &parent_span) {
+  const uint8_t *request_buffer_as_uint8 = reinterpret_cast<const uint8_t *>(req.request);
+  bool readOnly = req.flags & MsgFlag::READ_ONLY_FLAG;
+  if (readOnly) {
+    LOG_ERROR(m_logger,
+              "Received READ command for pre execution: "
+                  << "seqNum=" << req.executionSequenceNum << " batchCid" << batchCid);
+    return;
+  }
+  auto res = verifyWriteCommand(req.requestSize, request_buffer_as_uint8, req.maxReplySize, req.outActualReplySize);
+  if (res != OperationResult::SUCCESS) {
+    LOG_INFO(GL, "Operation result is not success in verifying write command");
+  } else {
+    SKVBCRequest deserialized_request;
+    deserialize(request_buffer_as_uint8, request_buffer_as_uint8 + req.requestSize, deserialized_request);
+    const SKVBCWriteRequest &write_req = std::get<SKVBCWriteRequest>(deserialized_request.request);
+    LOG_INFO(m_logger,
+             "Pre execute WRITE command:"
+                 << " type=SKVBCWriteRequest seqNum=" << req.executionSequenceNum
+                 << " numOfWrites=" << write_req.writeset.size() << " numOfKeysInReadSet=" << write_req.readset.size()
+                 << " readVersion=" << write_req.read_version);
+    if (write_req.long_exec) {
+      sleep(LONG_EXEC_CMD_TIME_IN_SEC);
+    }
+  }
+  req.outActualReplySize = req.requestSize;
+  memcpy(req.outReply, req.request, req.requestSize);
+  req.outExecutionStatus = static_cast<uint32_t>(res);
```

Comment:
This should be done only in case res.outExecutionStatus was initially set to UNKNOWN (see execute function).